lib/internal/http2/compat.js

```diff
@@ -550,13 +550,20 @@ class Http2ServerResponse extends Stream {
     if (headers === undefined && typeof statusMessage === 'object')
       headers = statusMessage;
 
-    if (typeof headers === 'object') {
+    let key;
+    if (typeof headers === 'object' && headers !== null) {
```

Comment:
Just to be fully clear, I'm against any change but adding `headers !== null` here and in http.